### ISSUE TITLE
Default _need_bytes min_wrong to 0x80 to avoid ByteWarning?

### ISSUE THREAD

**Issue by user202729** [Gallopsled/pwntools#2203]
Normally if `process.send` or something similar gets a string it gives the warning

```
BytesWarning: Bytes is not text; assuming ASCII, no guarantees.
```

While [the tutorial explains clearly what's the difference between bytes and string](https://github.com/Gallopsled/pwntools-tutorial/blob/master/bytes.md), I think it would be safe to suppress the warning in case the string consist of only characters of code < 0x80?

https://github.com/Gallopsled/pwntools/blob/f1693e3005089e4e169f8af1c765bb0e17ef8c09/pwnlib/util/packing.py#L1012

Currently `min_wrong` default to 0, I don't see any harm to change it to 0x80 because all of the (practical) encoding currently agrees with ASCII in that range anyway. (and it wouldn't affect the performance because the checks are already done)

**Comment by peace-maker** [comment]
I agree, this would make that warning less spammy for cases with ascii-only strings.

It could lead to inconsistent warnings when sending temporary data which happens to be printable in one run and not in another.

I don't know how much of this is "teaching" to always use bytes @Arusekk ?

**Comment by Arusekk** [comment]
My initial dillema was between warning and error, not between warning and silently doing unexpected stuff.

The ideal approach 'avoiding surprise' would be to actually find the source file, parse it, and raise some early SyntaxError if there was any e.g. '\xcc' escape in a text literal (forcing either b'\xcc' or '\u00cc' to avoid ambiguity), and only then warnings could be ignored IMO.

My ultimate goal was to teach everyone coming from C or from Perl that in Python you need to **always** add a b prefix when talking “octets” bytes, even if there are no high >'\x7f' literals.  And to be kind of type-strict internally by the way.

The bottom line: if someone finds the warnings annoying, they either switch often (above solution maybe?), or do not get the bytes vs text thing (and should definitely read further).

**Comment by peace-maker** [comment]
I'm fine with always requiring bytes. You have to understand the difference eventually and it's best to get used to prefixing strings with `b`. With those goals in mind, the `min_wrong` default value shouldn't be changed.
